### PR TITLE
Add support for nvim-lsp lsp-status

### PR DESCRIPTION
**Pull request by ray-x**
Lsp status support for eleline

### CODE REVIEW

File: plugin/eleline.vim [hunk 1]
```diff
@@ -230,9 +249,15 @@ function! s:StatusLine() abort
   let l:tags = '%{exists("b:gutentags_files") ? gutentags#statusline() : ""} '
   let l:lcn = '%{ElelineLCN()}'
   let l:coc = '%{ElelineCoc()}'
-  let l:vista = '%#ElelineVista#%{ElelineVista()}%*'
+  if empty(get(b:, 'vista_nearest_method_or_function', ''))
+    let l:lsp = '%{ElelineNvimLsp()}'
```
Comment (liuchengxu) [inline plugin/eleline.vim]: Since it's a neovim-only feature, we should check if we are using neovim.

File: plugin/eleline.vim [hunk 2]
```diff
@@ -204,6 +204,21 @@ function! ElelineVista() abort
   return !empty(get(b:, 'vista_nearest_method_or_function', '')) ? s:fn_icon.b:vista_nearest_method_or_function : ''
 endfunction
 
+function! ElelineNvimLsp() abort
+  if s:is_tmp_file()
+    return ''
+  endif
+  if luaeval('#vim.lsp.buf_get_clients() > 0')
+    let l:lsp_status = luaeval("require('lsp-status').status()")
```
Comment (liuchengxu) [inline plugin/eleline.vim:212]: I'm guessing neovim-nightly is required?

Comment (ray-x) [reply plugin/eleline.vim:212]: luaeval('#vim.lsp.buf_get_clients() > 0') will return 0 for vim and nvim-0.4

File: plugin/eleline.vim (hunk 1 above)
Comment (ray-x) [reply plugin/eleline.vim]: added has('nvim-0.5')

File: plugin/eleline.vim [hunk 3]
```diff
@@ -220,6 +235,10 @@ endfunction
 
 " https://github.com/liuchengxu/eleline.vim/wiki
 function! s:StatusLine() abort
+  if s:is_tmp_file()
+    return ''
+  endif
```
Comment (liuchengxu) [inline plugin/eleline.vim]: Why this?

Comment (ray-x) [reply plugin/eleline.vim]: It could be removed.  But a bit annoying and distracting to show the status bar for vista, defx, luatree etc.

Comment (liuchengxu) [reply plugin/eleline.vim]: Let's remove it and focus on the lsp support in this PR.

File: plugin/eleline.vim [hunk 4]
```diff
@@ -204,6 +204,21 @@ function! ElelineVista() abort
...
+  if luaeval('#vim.lsp.buf_get_clients() > 0')
+    let l:lsp_status = luaeval("require('lsp-status').status()")
+    if empty(l:lsp_status)
+      return ''
+    else
+      return s:fn_icon.l:lsp_status
+    endif
```
Comment (liuchengxu) [inline plugin/eleline.vim]:
```suggestion
    return empty(l:lsp_status) ? '' : s:fn_icon.l:lsp_status
```

File: plugin/eleline.vim (hunk 3 above)
Comment (ray-x) [reply plugin/eleline.vim]: Sure, done.

**Comment by liuchengxu** [comment]
Merged, thank you!

**Pull request merged**